docs/en/ingest-management/fleet/fleet-server-secrets.asciidoc

```diff
@@ -0,0 +1,252 @@
+See <<elastic-agent-cmd-options>> for more details.
+
+If running {fleet-server} under the {agent} in a container (see <<elastic-agent-container>>), the environment variables `FLEET_SERVER_SERVICE_TOKEN` or `FLEET_SERVER_SERVICE_TOKEN_PATH` may be used.
+
+TLS private key::
+The TLS private key is used to encrypt communications between {fleet-server} and {agent}.
+See <<secure-connections>> for more details.
+
+While not reccomended, it may be specified in the configuration directly as:
+[source,yaml]
+----
+inputs:
+  - type: fleet-server
+    ssl.key: |
+      ----BEGIN CERTIFICATE----
+      ....
+      ----END CERTIFICATE----
+----
+
+Alternatively, the path to the private key can be provided with the same attribute:
+[source,yaml]
+----
+inputs:
+  - type: fleet-server
+    ssl.key: /path/to/cert.key
+----
+
+When running {fleet-server} under {agent} the private key path may be provided with the `--fleet-server-cert-key` flag.
+See <<elastic-agent-cmd-options>> for more details.
+
+If running {fleet-server} under the {agent} in a container (see <<elastic-agent-container>>), the environment variable `FLEET_SERVER_CERT_KEY` may be used to specify the private key path.
+
+TLS private key passphrase::
+The private key passphrase is use to decrypt an encrypted private key file.
+
+It can be specfied as a secret file in the configuration with:
+[source,yaml]
+----
+inputs:
+  - type: fleet-server
+    ssl.key_passphrase_path: /path/to/passphrase
+----
+
+When running {fleet-server} under {agent} the passphrase path may be provided with the `--fleet-server-cert-key-passphrase-path` flag.
+See <<elastic-agent-cmd-options>> for more details.
+
+If running {fleet-server} under the {agent} in a container (see <<elastic-agent-container>>), the environment variable `FLEET_SERVER_CERT_KEY_PASSPHRASE` may be used to specify the file path.
+
+APM API Key::
+The APM API Key may be used to gather APM data from {fleet-server}.
+
+It can be directly specified in the instrumentation segment of the config:
+[source,yaml]
+----
+inputs:
+  - type: fleet-server
+    instrumentation.api_key: my-apm-api-key
+----
+
+Or by a file with:
+[source,yaml]
+----
+inputs:
+  - type: fleet-server
+    instrumentation.api_key_file: /path/to/apmAPIKey
+----
+
+It may also be specified by value with the environment variable `ELASTIC_APM_API_KEY`.
+
+APM secret token::
+The APM secret token may be used to gather APM data from {fleet-server}.
+
+It can be directly specified in the instrumentation segment of the config:
+[source,yaml]
+----
+inputs:
+  - type: fleet-server
+    instrumentation.secret_token: my-apm-secret-token
+----
+
+Or by a file with:
+[source,yaml]
+----
+inputs:
+  - type: fleet-server
+    instrumentation.secret_token_file: /path/to/apmSecretToken
+----
+
+It may also be specified by value with the environment variable `ELASTIC_APM_SECRET_TOKEN`.
+
+== Secret files guide
```

Comment:
```suggestion
[[secret-files-guide]]
== Secret files guide
```
This would creates an ID so that we can link to here from the top of the "Fleet Server Secrets" page.